lib/Optimizer/Transforms/ObserveAnsatz.cpp

```diff
@@ -90,6 +98,29 @@ struct AnsatzFunctionAnalysis {
         data.qubitValues.insert({op.getConstantIndex(), op.getResult()});
     });
 
+    // If mapping has moved qubits or introduced auxillary qubits, update the
+    // analysis accordingly.
+    if (auto mappingAttr =
+            dyn_cast_if_present<ArrayAttr>(funcOp->getAttr("mapping_v2p"))) {
```

Comment:
Out of curiosity, is the `dyn_cast_if_present` variant needed here? (I've been under the impression that `dyn_cast_or_null` was The Way.)

Reply:
The comment in `Casting.h` says that `dyn_cast_or_null` is deprecated.
```cpp
// Forwards to dyn_cast_if_present to avoid breaking current users. This is
// deprecated and will be removed in a future patch, use
// cast_if_present instead.
template <class X, class Y> auto dyn_cast_or_null(const Y &Val) {
  return dyn_cast_if_present<X>(Val);
}
```

Reply:
Thanks, Ben. We have a New Way, I guess.